stdlib/io.pyi

```diff
@@ -67,11 +68,7 @@ class FileIO(RawIOBase, BinaryIO):
     # Technically this is whatever is passed in as file, either a str, a bytes, or an int.
     name: Union[int, str]  # type: ignore
     def __init__(
-        self,
-        file: Union[str, bytes, int],
-        mode: str = ...,
-        closefd: bool = ...,
-        opener: Optional[Callable[[Union[int, str], str], int]] = ...,
+        self, file: Union[str, bytes, int], mode: str = ..., closefd: bool = ..., opener: Optional[_Opener] = ...
```

Comment:
I spent about 5 minutes calling `FileIO` in different ways, with `opener=print`, and couldn't get it to call it with `int` as first argument of `opener`. Now I'm convinced that this is correct :)

Nit about new-style unions:

```suggestion
        self, file: str | bytes | int, mode: str = ..., closefd: bool = ..., opener: _Opener | None = ...
```